### PR TITLE
chore: use component_owners

### PR DESCRIPTION
**Pull request by toddbaert**
Adds component_owners for all components

### CODE REVIEW

**Comment by beeme1mr** [comment]
The challenge with using a code owners file is that everyone listed needs write access to the repo. @dyladan created a tool that works around that limitation that we may want to consider.

**Comment by toddbaert** [comment]
> The challenge with using a code owners file is that everyone listed needs write access to the repo. @dyladan created a tool that works around that limitation that we may want to consider.

I've simply done this: https://github.com/open-feature/community/pull/221

This extracts the contrib roles from the sdk roles. With the CODEOWNERS in place, this should cover our use cases. CODEOWNERS will be able to approve changes to their own modules and merge them, but not release or make wider changes without SDK maintainer approval.

**Comment by dyladan** [comment]
> > The challenge with using a code owners file is that everyone listed needs write access to the repo. @dyladan created a tool that works around that limitation that we may want to consider.
> 
> I've simply done this: [open-feature/community#221](https://github.com/open-feature/community/pull/221)
> 
> This extracts the contrib roles from the sdk roles. With the CODEOWNERS in place, this should cover our use cases. CODEOWNERS will be able to approve changes to their own modules and merge them, but not release or make wider changes without SDK maintainer approval.

You're both right. The issue is that the contrib group can make changes to anything in the contrib repo or merge any PR that satisfies merge checks. If you're ok with that risk then I would recommend using the built in solution as long as it works for you. In otel we have many components with many different owners, some of whom are only occasionally present in the repo or are new to the repo so we wanted to be stricter with controlling those permissions.

**Comment by toddbaert** [comment]
> > > The challenge with using a code owners file is that everyone listed needs write access to the repo. @dyladan created a tool that works around that limitation that we may want to consider.
> > 
> > 
> > I've simply done this: [open-feature/community#221](https://github.com/open-feature/community/pull/221)
> > This extracts the contrib roles from the sdk roles. With the CODEOWNERS in place, this should cover our use cases. CODEOWNERS will be able to approve changes to their own modules and merge them, but not release or make wider changes without SDK maintainer approval.
> 
> You're both right. The issue is that the contrib group can make changes to anything in the contrib repo or merge any PR that satisfies merge checks. If you're ok with that risk then I would recommend using the built in solution as long as it works for you. In otel we have many components with many different owners, some of whom are only occasionally present in the repo or are new to the repo so we wanted to be stricter with controlling those permissions.

I'd be open to more robust tooling if it's not a lot of overhead. This is just what we'd already talked about. I guess the risk would be a "low trust" CODEOWNER pushing/approving changes that are not carefully reviewed when a maintainer later releases them?

**Comment by toddbaert** [comment]
I'll mark as draft for now.

**Comment by dyladan** [comment]
There are several potential risks of varying severity and likelihood, but most can be mitigated. The issue is that as the repo grows you have tens or hundreds of people with write permissions. Even if you've used branch protection rules and merge checks to lock things down as much as possible you may still run into issues like code owners merging PRs for components they don't own, closing issues that aren't resolved, et cetera. The more people have write permissions, the more likely it is something like this will happen even on accident.

**Comment by toddbaert** [comment]
> There are several potential risks of varying severity and likelihood, but most can be mitigated. The issue is that as the repo grows you have tens or hundreds of people with write permissions. Even if you've used branch protection rules and merge checks to lock things down as much as possible you may still run into issues like code owners merging PRs for components they don't own, closing issues that aren't resolved, et cetera. The more people have write permissions, the more likely it is something like this will happen even on accident.

Ok I'll look at the tooling you're referring to... Assuming I can find it in the otel js contribs?

**Comment by beeme1mr** [comment]
https://github.com/dyladan/component-owners

**Comment by toddbaert** [comment]
@dyladan @beeme1mr the TC has agrees to give this a go! Thanks Dan!

Please re-review.

File: .github/component_owners.yml
Comment (toddbaert) [inline .github/component_owners.yml:29]: I wasn't quite sure about this.

I guess we can always directly ping authors for renovate PRs that break or stay open for a long time.

File: .github/workflows/component-owners.yml
Comment (beeme1mr) [inline .github/workflows/component-owners.yml:10]: Do you think we should pin it to a sha or tag?

**Review by beeme1mr — APPROVED** [review]

**Review by sago2k8 — APPROVED** [review]
LGTM, I'm ok being pinned for maintaining the provider.

thank you

File: .github/workflows/component-owners.yml
Comment (toddbaert) [reply .github/workflows/component-owners.yml:10]: I considered it, but this is what Dan has in OTel so I went this way.

Comment (kinyoklion) [reply .github/workflows/component-owners.yml:10]: I'd also be comfortable with slightly less mutability, or at least targeted mutability (like a tag).

Or reducing the permissions for these jobs specifically to what they need to decrease the supply chain risk.
https://docs.github.com/en/actions/using-jobs/assigning-permissions-to-jobs

Comment (toddbaert) [reply .github/workflows/component-owners.yml]: Good points, see [here](https://github.com/open-feature/js-sdk-contrib/pull/446/files#r1259750922).

File: .github/component_owners.yml
Comment (lukas-reining) [reply .github/component_owners.yml:7]: Looks good to me thanks!

**Review by lukas-reining — APPROVED** [review]

**Review by kinyoklion — APPROVED** [review]

**Review by beeme1mr — APPROVED** [review]

**Pull request merged**
